### ISSUE TITLE
Custom columns failing when joined data has over about 30 columns

### ISSUE THREAD

**Issue by notrom** [metabase/metabase#12481]
**Describe the bug**
When trying to create a custom column after joining tables with many columns the custom column preview and whole question fails.

**Logs**
No javascript console errors.

Server log:
```
[d5310881-11d9-42ff-94cb-fac2d2ef9d1a] 2020-05-08T14:36:03+12:00 ERROR metabase.query-processor.middleware.catch-exceptions Error processing query: null
{:database_id 1,
:started_at (t/zoned-date-time "2020-05-08T14:36:02.674+12:00[Pacific/Auckland]"),
:state "42S22",
:json_query
{:type "query",
:query
{:source-table 2,
:expressions {:foo ["*" ["field-id" 14] ["field-id" 13]]},
:joins
[{:fields "all", :source-table 1, :condition ["=" ["field-id" 11] ["joined-field" "Product" ["field-id" 4]]], :alias "Product"}
{:fields "all", :source-table 4, :condition ["=" ["joined-field" "Product" ["field-id" 4]] ["joined-field" "Reviews" ["field-id" 31]]], :alias "Reviews"}
{:fields "all", :source-table 3, :condition ["=" ["field-id" 9] ["joined-field" "User" ["field-id" 21]]], :alias "User"}],
:limit 10},
:database 1,
:parameters [],
:middleware {:add-default-userland-constraints? true}},
:native
{:query
"SELECT \"source\".\"ID\" AS \"ID\", \"source\".\"CREATED_AT\" AS \"CREATED_AT\", \"source\".\"DISCOUNT\" AS \"DISCOUNT\", \"source\".\"PRODUCT_ID\" AS \"PRODUCT_ID\", \"source\".\"QUANTITY\" AS \"QUANTITY\", \"source\".\"SUBTOTAL\" AS \"SUBTOTAL\", \"source\".\"TAX\" AS \"TAX\", \"source\".\"TOTAL\" AS \"TOTAL\", \"source\".\"USER_ID\" AS \"USER_ID\", \"source\".\"foo\" AS \"foo\", \"source\".\"ID_2\" AS \"ID_2\", \"source\".\"CATEGORY\" AS \"CATEGORY\", \"source\".\"CREATED_AT_2\" AS \"CREATED_AT_2\", \"source\".\"EAN\" AS \"EAN\", \"source\".\"PRICE\" AS \"PRICE\", \"source\".\"RATING\" AS \"RATING\", \"source\".\"TITLE\" AS \"TITLE\", \"source\".\"VENDOR\" AS \"VENDOR\", \"source\".\"ID_3\" AS \"ID_3\", \"source\".\"BODY\" AS \"BODY\", \"source\".\"CREATED_AT_3\" AS \"CREATED_AT_3\", \"source\".\"PRODUCT_ID_2\" AS \"PRODUCT_ID_2\", \"source\".\"RATING_2\" AS \"RATING_2\", \"source\".\"REVIEWER\" AS \"REVIEWER\", \"source\".\"ID_4\" AS \"ID_4\", \"source\".\"NAME\" AS \"NAME\", \"source\".\"ADDRESS\" AS \"ADDRESS\", \"source\".\"BIRTH_DATE\" AS \"BIRTH_DATE\", \"source\".\"CITY\" AS \"CITY\", \"source\".\"CREATED_AT_4\" AS \"CREATED_AT_4\", \"source\".\"EMAIL\" AS \"EMAIL\", \"source\".\"LATITUDE\" AS \"LATITUDE\", \"PUBLIC\".\"PEOPLE\".\"LONGITUDE\" AS \"LONGITUDE\", \"PUBLIC\".\"PEOPLE\".\"PASSWORD\" AS \"PASSWORD\", \"PUBLIC\".\"PEOPLE\".\"SOURCE\" AS \"SOURCE\", \"PUBLIC\".\"PEOPLE\".\"STATE\" AS \"STATE\", \"PUBLIC\".\"PEOPLE\".\"ZIP\" AS \"ZIP\" FROM (SELECT (\"PUBLIC\".\"ORDERS\".\"QUANTITY\" * \"PUBLIC\".\"ORDERS\".\"SUBTOTAL\") AS \"foo\", \"PUBLIC\".\"ORDERS\".\"ID\" AS \"ID\", \"PUBLIC\".\"ORDERS\".\"CREATED_AT\" AS \"CREATED_AT\", \"PUBLIC\".\"ORDERS\".\"DISCOUNT\" AS \"DISCOUNT\", \"PUBLIC\".\"ORDERS\".\"PRODUCT_ID\" AS \"PRODUCT_ID\", \"PUBLIC\".\"ORDERS\".\"QUANTITY\" AS \"QUANTITY\", \"PUBLIC\".\"ORDERS\".\"SUBTOTAL\" AS \"SUBTOTAL\", \"PUBLIC\".\"ORDERS\".\"TAX\" AS \"TAX\", \"PUBLIC\".\"ORDERS\".\"TOTAL\" AS \"TOTAL\", \"PUBLIC\".\"ORDERS\".\"USER_ID\" AS \"USER_ID\", \"Product\".\"ID\" AS \"ID_2\", \"Product\".\"CATEGORY\" AS \"CATEGORY\", \"Product\".\"CREATED_AT\" AS \"CREATED_AT_2\", \"Product\".\"EAN\" AS \"EAN\", \"Product\".\"PRICE\" AS \"PRICE\", \"Product\".\"RATING\" AS \"RATING\", \"Product\".\"TITLE\" AS \"TITLE\", \"Product\".\"VENDOR\" AS \"VENDOR\", \"Reviews\".\"ID\" AS \"ID_3\", \"Reviews\".\"BODY\" AS \"BODY\", \"Reviews\".\"CREATED_AT\" AS \"CREATED_AT_3\", \"Reviews\".\"PRODUCT_ID\" AS \"PRODUCT_ID_2\", \"Reviews\".\"RATING\" AS \"RATING_2\", \"Reviews\".\"REVIEWER\" AS \"REVIEWER\", \"User\".\"ID\" AS \"ID_4\", \"User\".\"NAME\" AS \"NAME\", \"User\".\"ADDRESS\" AS \"ADDRESS\", \"User\".\"BIRTH_DATE\" AS \"BIRTH_DATE\", \"User\".\"CITY\" AS \"CITY\", \"User\".\"CREATED_AT\" AS \"CREATED_AT_4\", \"User\".\"EMAIL\" AS \"EMAIL\", \"User\".\"LATITUDE\" AS \"LATITUDE\", \"User\".\"LONGITUDE\" AS \"LONGITUDE\", \"User\".\"PASSWORD\" AS \"PASSWORD\", \"User\".\"SOURCE\" AS \"SOURCE\", \"User\".\"STATE\" AS \"STATE\", \"User\".\"ZIP\" AS \"ZIP\" FROM \"PUBLIC\".\"ORDERS\" LEFT JOIN \"PUBLIC\".\"PRODUCTS\" \"Product\" ON \"PUBLIC\".\"ORDERS\".\"PRODUCT_ID\" = \"Product\".\"ID\" LEFT JOIN \"PUBLIC\".\"REVIEWS\" \"Reviews\" ON \"Product\".\"ID\" = \"Reviews\".\"PRODUCT_ID\" LEFT JOIN \"PUBLIC\".\"PEOPLE\" \"User\" ON \"PUBLIC\".\"ORDERS\".\"USER_ID\" = \"User\".\"ID\") \"source\" LIMIT 10",
:params nil},
:status :failed,
:class org.h2.jdbc.JdbcSQLException,
:stacktrace
["org.h2.message.DbException.getJdbcSQLException(DbException.java:357)"
"org.h2.message.DbException.get(DbException.java:179)"
"org.h2.message.DbException.get(DbException.java:155)"
"org.h2.expression.ExpressionColumn.optimize(ExpressionColumn.java:150)"
"org.h2.expression.Alias.optimize(Alias.java:51)"
"org.h2.command.dml.Select.prepare(Select.java:858)"
"org.h2.command.Parser.prepareCommand(Parser.java:283)"
"org.h2.engine.Session.prepareLocal(Session.java:611)"
"org.h2.engine.Session.prepareCommand(Session.java:549)"
"org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)"
"org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:76)"
"org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:1135)"
"com.mchange.v2.c3p0.impl.NewProxyConnection.prepareStatement(NewProxyConnection.java:267)"
"--> driver.sql_jdbc.execute$fn__70587.invokeStatic(execute.clj:239)"
"driver.sql_jdbc.execute$fn__70587.invoke(execute.clj:237)"
"driver.sql_jdbc.execute$prepared_statement_STAR_.invokeStatic(execute.clj:257)"
"driver.sql_jdbc.execute$prepared_statement_STAR_.invoke(execute.clj:254)"
"driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:387)"
"driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:377)"
"driver.sql_jdbc$fn__72711.invokeStatic(sql_jdbc.clj:50)"
"driver.sql_jdbc$fn__72711.invoke(sql_jdbc.clj:48)"
"driver.h2$fn__71677.invokeStatic(h2.clj:81)"
"driver.h2$fn__71677.invoke(h2.clj:78)"
"query_processor.context$executef.invokeStatic(context.clj:59)"
"query_processor.context$executef.invoke(context.clj:48)"
"query_processor.context.default$default_runf.invokeStatic(default.clj:69)"
"query_processor.context.default$default_runf.invoke(default.clj:67)"
"query_processor.context$runf.invokeStatic(context.clj:45)"
"query_processor.context$runf.invoke(context.clj:39)"
"query_processor.reducible$pivot.invokeStatic(reducible.clj:34)"
"query_processor.reducible$pivot.invoke(reducible.clj:31)"
"query_processor.middleware.mbql_to_native$mbql__GT_native$fn__43003.invoke(mbql_to_native.clj:26)"
"query_processor.middleware.check_features$check_features$fn__42317.invoke(check_features.clj:42)"
"query_processor.middleware.optimize_datetime_filters$optimize_datetime_filters$fn__43168.invoke(optimize_datetime_filters.clj:133)"
"query_processor.middleware.wrap_value_literals$wrap_value_literals$fn__47065.invoke(wrap_value_literals.clj:137)"
"query_processor.middleware.annotate$add_column_info$fn__40946.invoke(annotate.clj:577)"
"query_processor.middleware.permissions$check_query_permissions$fn__42192.invoke(permissions.clj:64)"
"query_processor.middleware.pre_alias_aggregations$pre_alias_aggregations$fn__43667.invoke(pre_alias_aggregations.clj:40)"
"query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__42390.invoke(cumulative_aggregations.clj:61)"
"query_processor.middleware.resolve_joins$resolve_joins$fn__44199.invoke(resolve_joins.clj:183)"
"query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__39133.invoke(add_implicit_joins.clj:245)"
"query_processor.middleware.limit$limit$fn__42989.invoke(limit.clj:38)"
"query_processor.middleware.format_rows$format_rows$fn__42970.invoke(format_rows.clj:81)"
"query_processor.middleware.desugar$desugar$fn__42456.invoke(desugar.clj:22)"
"query_processor.middleware.binning$update_binning_strategy$fn__41490.invoke(binning.clj:229)"
"query_processor.middleware.resolve_fields$resolve_fields$fn__41998.invoke(resolve_fields.clj:24)"
"query_processor.middleware.add_dimension_projections$add_remapping$fn__38669.invoke(add_dimension_projections.clj:272)"
"query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__38889.invoke(add_implicit_clauses.clj:147)"
"query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__39282.invoke(add_source_metadata.clj:105)"
"query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__43864.invoke(reconcile_breakout_and_order_by_bucketing.clj:98)"
"query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__41131.invoke(auto_bucket_datetimes.clj:125)"
"query_processor.middleware.resolve_source_table$resolve_source_tables$fn__42045.invoke(resolve_source_table.clj:46)"
"query_processor.middleware.parameters$substitute_parameters$fn__43649.invoke(parameters.clj:97)"
"query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__42097.invoke(resolve_referenced.clj:80)"
"query_processor.middleware.expand_macros$expand_macros$fn__42712.invoke(expand_macros.clj:158)"
"query_processor.middleware.add_timezone_info$add_timezone_info$fn__39313.invoke(add_timezone_info.clj:15)"
"query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__46949.invoke(splice_params_in_response.clj:32)"
"query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__43875$fn__43879.invoke(resolve_database_and_driver.clj:33)"
"driver$do_with_driver.invokeStatic(driver.clj:61)"
"driver$do_with_driver.invoke(driver.clj:57)"
"query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__43875.invoke(resolve_database_and_driver.clj:27)"
"query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__42918.invoke(fetch_source_query.clj:243)"
"query_processor.middleware.store$initialize_store$fn__46958$fn__46959.invoke(store.clj:11)"
"query_processor.store$do_with_store.invokeStatic(store.clj:46)"
"query_processor.store$do_with_store.invoke(store.clj:40)"
"query_processor.middleware.store$initialize_store$fn__46958.invoke(store.clj:10)"
"query_processor.middleware.cache$maybe_return_cached_results$fn__41974.invoke(cache.clj:208)"
"query_processor.middleware.validate$validate_query$fn__46967.invoke(validate.clj:10)"
"query_processor.middleware.normalize_query$normalize$fn__43016.invoke(normalize_query.clj:22)"
"query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__39151.invoke(add_rows_truncated.clj:36)"
"query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__46934.invoke(results_metadata.clj:128)"
"query_processor.middleware.constraints$add_default_userland_constraints$fn__42333.invoke(constraints.clj:42)"
"query_processor.middleware.process_userland_query$process_userland_query$fn__43738.invoke(process_userland_query.clj:136)"
"query_processor.middleware.catch_exceptions$catch_exceptions$fn__42276.invoke(catch_exceptions.clj:174)"
"query_processor.reducible$async_qp$qp_STAR___37952$thunk__37953.invoke(reducible.clj:101)"
"query_processor.reducible$async_qp$qp_STAR___37952.invoke(reducible.clj:107)"
"query_processor.reducible$sync_qp$qp_STAR___37961$fn__37964.invoke(reducible.clj:133)"
"query_processor.reducible$sync_qp$qp_STAR___37961.invoke(reducible.clj:132)"
"query_processor$process_userland_query.invokeStatic(query_processor.clj:218)"
"query_processor$process_userland_query.doInvoke(query_processor.clj:214)"
"query_processor$fn__47133$process_query_and_save_execution_BANG___47142$fn__47145.invoke(query_processor.clj:233)"
"query_processor$fn__47133$process_query_and_save_execution_BANG___47142.invoke(query_processor.clj:225)"
"query_processor$fn__47177$process_query_and_save_with_max_results_constraints_BANG___47186$fn__47189.invoke(query_processor.clj:245)"
"query_processor$fn__47177$process_query_and_save_with_max_results_constraints_BANG___47186.invoke(query_processor.clj:238)"
"api.dataset$fn__50475$fn__50478.invoke(dataset.clj:52)"
"query_processor.streaming$streaming_response_STAR_$fn__33990$fn__33991.invoke(streaming.clj:73)"
"query_processor.streaming$streaming_response_STAR_$fn__33990.invoke(streaming.clj:72)"
"async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:66)"
"async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:64)"
"async.streaming_response$do_f_async$fn__23079.invoke(streaming_response.clj:85)"],
:context :ad-hoc,
:error
"Column \"PUBLIC.PEOPLE.LONGITUDE\" not found; SQL statement:\n-- Metabase:: userID: 1 queryType: MBQL queryHash: b5522855682c63e071eeb44168ec45332bc3b6e0a06b81d22d324c44ca91d535\nSELECT \"source\".\"ID\" AS \"ID\", \"source\".\"CREATED_AT\" AS \"CREATED_AT\", \"source\".\"DISCOUNT\" AS \"DISCOUNT\", \"source\".\"PRODUCT_ID\" AS \"PRODUCT_ID\", \"source\".\"QUANTITY\" AS \"QUANTITY\", \"source\".\"SUBTOTAL\" AS \"SUBTOTAL\", \"source\".\"TAX\" AS \"TAX\", \"source\".\"TOTAL\" AS \"TOTAL\", \"source\".\"USER_ID\" AS \"USER_ID\", \"source\".\"foo\" AS \"foo\", \"source\".\"ID_2\" AS \"ID_2\", \"source\".\"CATEGORY\" AS \"CATEGORY\", \"source\".\"CREATED_AT_2\" AS \"CREATED_AT_2\", \"source\".\"EAN\" AS \"EAN\", \"source\".\"PRICE\" AS \"PRICE\", \"source\".\"RATING\" AS \"RATING\", \"source\".\"TITLE\" AS \"TITLE\", \"source\".\"VENDOR\" AS \"VENDOR\", \"source\".\"ID_3\" AS \"ID_3\", \"source\".\"BODY\" AS \"BODY\", \"source\".\"CREATED_AT_3\" AS \"CREATED_AT_3\", \"source\".\"PRODUCT_ID_2\" AS \"PRODUCT_ID_2\", \"source\".\"RATING_2\" AS \"RATING_2\", \"source\".\"REVIEWER\" AS \"REVIEWER\", \"source\".\"ID_4\" AS \"ID_4\", \"source\".\"NAME\" AS \"NAME\", \"source\".\"ADDRESS\" AS \"ADDRESS\", \"source\".\"BIRTH_DATE\" AS \"BIRTH_DATE\", \"source\".\"CITY\" AS \"CITY\", \"source\".\"CREATED_AT_4\" AS \"CREATED_AT_4\", \"source\".\"EMAIL\" AS \"EMAIL\", \"source\".\"LATITUDE\" AS \"LATITUDE\", \"PUBLIC\".\"PEOPLE\".\"LONGITUDE\" AS \"LONGITUDE\", \"PUBLIC\".\"PEOPLE\".\"PASSWORD\" AS \"PASSWORD\", \"PUBLIC\".\"PEOPLE\".\"SOURCE\" AS \"SOURCE\", \"PUBLIC\".\"PEOPLE\".\"STATE\" AS \"STATE\", \"PUBLIC\".\"PEOPLE\".\"ZIP\" AS \"ZIP\" FROM (SELECT (\"PUBLIC\".\"ORDERS\".\"QUANTITY\" * \"PUBLIC\".\"ORDERS\".\"SUBTOTAL\") AS \"foo\", \"PUBLIC\".\"ORDERS\".\"ID\" AS \"ID\", \"PUBLIC\".\"ORDERS\".\"CREATED_AT\" AS \"CREATED_AT\", \"PUBLIC\".\"ORDERS\".\"DISCOUNT\" AS \"DISCOUNT\", \"PUBLIC\".\"ORDERS\".\"PRODUCT_ID\" AS \"PRODUCT_ID\", \"PUBLIC\".\"ORDERS\".\"QUANTITY\" AS \"QUANTITY\", \"PUBLIC\".\"ORDERS\".\"SUBTOTAL\" AS \"SUBTOTAL\", \"PUBLIC\".\"ORDERS\".\"TAX\" AS \"TAX\", \"PUBLIC\".\"ORDERS\".\"TOTAL\" AS \"TOTAL\", \"PUBLIC\".\"ORDERS\".\"USER_ID\" AS \"USER_ID\", \"Product\".\"ID\" AS \"ID_2\", \"Product\".\"CATEGORY\" AS \"CATEGORY\", \"Product\".\"CREATED_AT\" AS \"CREATED_AT_2\", \"Product\".\"EAN\" AS \"EAN\", \"Product\".\"PRICE\" AS \"PRICE\", \"Product\".\"RATING\" AS \"RATING\", \"Product\".\"TITLE\" AS \"TITLE\", \"Product\".\"VENDOR\" AS \"VENDOR\", \"Reviews\".\"ID\" AS \"ID_3\", \"Reviews\".\"BODY\" AS \"BODY\", \"Reviews\".\"CREATED_AT\" AS \"CREATED_AT_3\", \"Reviews\".\"PRODUCT_ID\" AS \"PRODUCT_ID_2\", \"Reviews\".\"RATING\" AS \"RATING_2\", \"Reviews\".\"REVIEWER\" AS \"REVIEWER\", \"User\".\"ID\" AS \"ID_4\", \"User\".\"NAME\" AS \"NAME\", \"User\".\"ADDRESS\" AS \"ADDRESS\", \"User\".\"BIRTH_DATE\" AS \"BIRTH_DATE\", \"User\".\"CITY\" AS \"CITY\", \"User\".\"CREATED_AT\" AS \"CREATED_AT_4\", \"User\".\"EMAIL\" AS \"EMAIL\", \"User\".\"LATITUDE\" AS \"LATITUDE\", \"User\".\"LONGITUDE\" AS \"LONGITUDE\", \"User\".\"PASSWORD\" AS \"PASSWORD\", \"User\".\"SOURCE\" AS \"SOURCE\", \"User\".\"STATE\" AS \"STATE\", \"User\".\"ZIP\" AS \"ZIP\" FROM \"PUBLIC\".\"ORDERS\" LEFT JOIN \"PUBLIC\".\"PRODUCTS\" \"Product\" ON \"PUBLIC\".\"ORDERS\".\"PRODUCT_ID\" = \"Product\".\"ID\" LEFT JOIN \"PUBLIC\".\"REVIEWS\" \"Reviews\" ON \"Product\".\"ID\" = \"Reviews\".\"PRODUCT_ID\" LEFT JOIN \"PUBLIC\".\"PEOPLE\" \"User\" ON \"PUBLIC\".\"ORDERS\".\"USER_ID\" = \"User\".\"ID\") \"source\" LIMIT 10 [42122-197]",
:row_count 0,
:running_time 0,
:preprocessed
{:type :query,
:query
{:source-table 2,
:expressions {:foo [:* [:field-id 14] [:field-id 13]]},
:joins
[{:strategy :left-join, :source-table 1, :condition [:= [:field-id 11] [:joined-field "Product" [:field-id 4]]], :alias "Product"}
{:strategy :left-join, :source-table 4, :condition [:= [:joined-field "Product" [:field-id 4]] [:joined-field "Reviews" [:field-id 31]]], :alias "Reviews"}
{:strategy :left-join, :source-table 3, :condition [:= [:field-id 9] [:joined-field "User" [:field-id 21]]], :alias "User"}],
:limit 10,
:fields
[[:field-id 12]
[:datetime-field [:field-id 15] :default]
[:field-id 10]
[:field-id 11]
[:field-id 14]
[:field-id 13]
[:field-id 16]
[:field-id 17]
[:field-id 9]
[:expression "foo"]
[:joined-field "Product" [:field-id 4]]
[:joined-field "Product" [:field-id 6]]
[:datetime-field [:joined-field "Product" [:field-id 7]] :default]
[:joined-field "Product" [:field-id 1]]
[:joined-field "Product" [:field-id 3]]
[:joined-field "Product" [:field-id 2]]
[:joined-field "Product" [:field-id 5]]
[:joined-field "Product" [:field-id 8]]
[:joined-field "Reviews" [:field-id 32]]
[:joined-field "Reviews" [:field-id 33]]
[:datetime-field [:joined-field "Reviews" [:field-id 35]] :default]
[:joined-field "Reviews" [:field-id 31]]
[:joined-field "Reviews" [:field-id 36]]
[:joined-field "Reviews" [:field-id 34]]
[:joined-field "User" [:field-id 21]]
[:joined-field "User" [:field-id 20]]
[:joined-field "User" [:field-id 22]]
[:datetime-field [:joined-field "User" [:field-id 19]] :default]
[:joined-field "User" [:field-id 30]]
[:datetime-field [:joined-field "User" [:field-id 26]] :default]
[:joined-field "User" [:field-id 25]]
[:joined-field "User" [:field-id 18]]
[:joined-field "User" [:field-id 23]]
[:joined-field "User" [:field-id 29]]
[:joined-field "User" [:field-id 24]]
[:joined-field "User" [:field-id 28]]
[:joined-field "User" [:field-id 27]]]},
:database 1,
:middleware {:add-default-userland-constraints? true},
:info
{:executed-by 1,
:context :ad-hoc,
:nested? false,
:query-hash [-75, 82, 40, 85, 104, 44, 99, -32, 113, -18, -76, 65, 104, -20, 69, 51, 43, -61, -74, -32, -96, 107, -127, -46, 45, 50, 76, 68, -54, -111, -43, 53]},
:constraints {:max-results 10000, :max-results-bare-rows 2000}},
:data {:rows [], :cols []}}
```

**To Reproduce**
1. New Question > Sample Dataset > Orders > Query editor
2. Join Products on Product ID = ID
3. Join Reviews on ID = Product ID
4. Join People on User ID = ID
5. Preview after the final join, all good so far, results in 36 columns in the result set
6. Create Custom column e.g. "[Quantity] * [Subtotal]"
7. Preview custom column, this will fail and give and error
8. On any of the joined tables untick about 6 columns so they are not included in the output. You can use pretty much any combination of columns.
9. Preview the custom column and the query will work (although sometimes the new column itself isn't actually included in the output but you can use it in summaries etc.)

**Expected behavior**
I expect to be able to create custom columns on any joined set of data in the editor.

**Screenshots**
![image](https://user-images.githubusercontent.com/4504437/81370597-a845ac80-9149-11ea-906e-8e859da175ad.png)

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-NZ",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.129 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "Cp1252",
    "java.runtime.name": "Java(TM) SE Runtime Environment",
    "java.runtime.version": "1.8.0_161-b12",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "1.8.0_161",
    "java.vm.name": "Java HotSpot(TM) Client VM",
    "java.vm.version": "25.161-b12",
    "os.name": "Windows Server 2012 R2",
    "os.version": "6.3",
    "user.language": "en",
    "user.timezone": "Pacific/Auckland"
  },
  "metabase-info": {
    "databases": [
      "sqlserver",
      "postgres",
      "h2",
      "csv"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "10.5"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "tag": "v0.35.3",
      "date": "2020-04-21",
      "branch": "release-0.35.x",
      "hash": "1d424cb"
    },
    "settings": {
      "report-timezone": "Pacific/Auckland"
    }
  }
}
```

**Severity**
This is the second bug I've encountered in the custom column area today that has just stopped me in my tracks, now I know I can usually work around it by excluding a bunch of columns it's not as bad, but it did lead to a lot of confusion and frustration.

**Additional context**
It's also impacting my queries against SQL Server and Postgres (against the Metabase database itself), so it's not just an H2 thing. It seems to be if the joined result set exceeds around about 32 columns then custom columns won't work. This number doesn't seem to be totally consistent though.

**Comment by notrom** [comment]
Taking a look and it doesn't even need to be a joined dataset. I've got tables with way more than 30 columns and just using one of them I can't create a custom column until I exclude columns until it's 30 or less.

SQL Server give this error:
`The multi-part identifier "rel.UserDetails.UserName" could not be bound.`

![image](https://user-images.githubusercontent.com/4504437/81371668-8c8fd580-914c-11ea-9834-51f0af0d0b9f.png)

**Comment by sbelak** [comment]
I'm pretty sure this is the same bug as was fixed in #12387 and will be included in the next point release.

**Comment by flamber** [comment]
Confirmed that it is fixed by laziness fixes - I get errors on 0.35.3, but works on `release-0.35.x`
